### PR TITLE
[Merged by Bors] - feat(Order/SuccPred/Limit): induction principles for `IsSuccLimit` and `IsPredLimit`

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/Limit.lean [hunk 1]
```diff
@@ -654,21 +637,99 @@ theorem isPredPrelimitRecOn_pred' (hb : ¬ IsMin b) :
...
+variable (b) in
+open Classical in
+/-- A value can be built by building it on minimal elements, successors, and successor limits. -/
+@[elab_as_elim]
+noncomputable def isSuccLimitRecOn : C b :=
+  isSuccPrelimitRecOn b hs fun a ha ↦
+    if h : IsMin a then hm a h else hl a (ha.isSuccLimit_of_not_isMin h)
+
+@[simp]
+theorem isSuccLimitRecOn_limit (hb : IsSuccLimit b) : isSuccLimitRecOn b hm hs hl = hl b hb := by
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Limit.lean:665]: Maybe take the opportunity to fix this name and `isSuccPrelimitRecOn_limit`?
```suggestion
theorem isSuccLimitRecOn_of_isSuccLimit (hb : IsSuccLimit b) : isSuccLimitRecOn b hm hs hl = hl b hb := by
```

File: Mathlib/Order/SuccPred/Limit.lean [hunk 2]
```diff
@@ -654,21 +637,99 @@ theorem isPredPrelimitRecOn_pred' (hb : ¬ IsMin b) :
...
+    isSuccLimitRecOn (succ b) hm hs hl = hs b hb := by
+  rw [isSuccLimitRecOn, isSuccPrelimitRecOn_succ']
+
+@[simp]
+theorem isSuccLimitRecOn_succ [NoMaxOrder α] (b : α) :
+    isSuccLimitRecOn (succ b) hm hs hl = hs b (not_isMax b) :=
+  isSuccLimitRecOn_succ' hm hs hl _
+
+@[simp]
+theorem isSuccLimitRecOn_isMin (hb : IsMin b) : isSuccLimitRecOn b hm hs hl = hm b hb := by
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Limit.lean:685]:
```suggestion
theorem isSuccLimitRecOn_of_isMin (hb : IsMin b) : isSuccLimitRecOn b hm hs hl = hm b hb := by
```

File: Mathlib/Order/SuccPred/Limit.lean [hunk 3]
```diff
@@ -654,21 +637,99 @@ theorem isPredPrelimitRecOn_pred' (hb : ¬ IsMin b) :
...
+  isSuccLimitRecOn_limit (α := αᵒᵈ) hm hs _ hb.dual
+
+end PartialOrder
+
+section LinearOrder
+
+variable [LinearOrder α] [PredOrder α]
+  (hm : ∀ a, IsMax a → C a) (hs : ∀ a, ¬ IsMin a → C (pred a)) (hl : ∀ a, IsPredLimit a → C a)
+
+theorem isPredLimitRecOn_pred' (hb : ¬ IsMin b) : isPredLimitRecOn (pred b) hm hs hl = hs b hb :=
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Limit.lean:717]:
```suggestion
theorem isPredLimitRecOn_pred_of_not_isMin (hb : ¬ IsMin b) : isPredLimitRecOn (pred b) hm hs hl = hs b hb :=
```

File: Mathlib/Order/SuccPred/Limit.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/Order/SuccPred/Limit.lean:665]: I don't like how wordy the new names are, but I guess they are more consistent.

**Comment by YaelDillies** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**